Backend/lib/payment/src/Lib/Payment/Storage/Queries/PaymentTransaction.hs

```diff
@@ -11,67 +11,72 @@
-        PaymentTransactionMandateEndDate =. val transaction.mandateEndDate,
-        PaymentTransactionMandateId =. val transaction.mandateId,
-        PaymentTransactionMandateFrequency =. val transaction.mandateFrequency,
-        PaymentTransactionMandateMaxAmount =. val transaction.mandateMaxAmount,
-        PaymentTransactionUpdatedAt =. val now
-      ]
-    where_ $ tbl ^. PaymentTransactionId ==. val transaction.id.getId
+  updateWithKV
+    [ Se.Set BeamPT.statusId transaction.statusId,
+      Se.Set BeamPT.status transaction.status,
+      Se.Set BeamPT.paymentMethodType transaction.paymentMethodType,
+      Se.Set BeamPT.paymentMethod transaction.paymentMethod,
+      Se.Set BeamPT.respMessage transaction.respMessage,
+      Se.Set BeamPT.respCode transaction.respCode,
+      Se.Set BeamPT.gatewayReferenceId transaction.gatewayReferenceId,
+      Se.Set BeamPT.amount transaction.amount,
+      Se.Set BeamPT.currency transaction.currency,
+      Se.Set BeamPT.juspayResponse transaction.juspayResponse,
+      Se.Set BeamPT.mandateStatus transaction.mandateStatus,
+      Se.Set BeamPT.mandateStartDate transaction.mandateStartDate,
+      Se.Set BeamPT.mandateEndDate transaction.mandateEndDate,
+      Se.Set BeamPT.mandateId transaction.mandateId,
+      Se.Set BeamPT.mandateFrequency transaction.mandateFrequency,
+      Se.Set BeamPT.mandateMaxAmount transaction.mandateMaxAmount,
+      Se.Set BeamPT.updatedAt now
+    ]
+    [Se.Is BeamPT.id $ Se.Eq $ getId transaction.id]
 
-findByTxnUUID :: Transactionable m => Text -> m (Maybe PaymentTransaction)
-findByTxnUUID txnUUID =
-  findOne $ do
-    transaction <- from $ table @PaymentTransactionT
-    where_ $ transaction ^. PaymentTransactionTxnUUID ==. val (Just txnUUID)
-    return transaction
+findByTxnUUID :: BeamFlow m => Text -> m (Maybe PaymentTransaction)
+findByTxnUUID txnUUID = findOneWithKV [Se.Is BeamPT.txnUUID $ Se.Eq $ Just txnUUID]
 
-findAllByOrderId :: Transactionable m => Id PaymentOrder -> m [PaymentTransaction]
-findAllByOrderId orderId =
-  findAll $ do
-    transaction <- from $ table @PaymentTransactionT
-    where_ $ transaction ^. PaymentTransactionOrderId ==. val (toKey orderId)
-    orderBy [desc $ transaction ^. PaymentTransactionCreatedAt]
-    return transaction
+findAllByOrderId :: BeamFlow m => Id PaymentOrder -> m [PaymentTransaction]
+findAllByOrderId (Id orderId) = findAllWithOptionsKV [Se.Is BeamPT.orderId $ Se.Eq orderId] (Se.Desc BeamPT.createdAt) Nothing Nothing
 
-findNewTransactionByOrderId :: Transactionable m => Id PaymentOrder -> m (Maybe PaymentTransaction)
-findNewTransactionByOrderId orderId =
-  findOne $ do
-    transaction <- from $ table @PaymentTransactionT
-    where_ $
-      Esq.isNothing (transaction ^. PaymentTransactionTxnUUID)
-        &&. transaction ^. PaymentTransactionOrderId ==. val (toKey orderId)
-    limit 1
-    return transaction
+findNewTransactionByOrderId :: BeamFlow m => Id PaymentOrder -> m (Maybe PaymentTransaction)
```

Comment:
Its better to format queries like this for easy readability

```
findWindowsWithFeeType :: MonadFlow m => Id Merchant -> UTCTime -> UTCTime -> FeeType -> m [DriverFee]
findWindowsWithFeeType merchantId from to feeType =
  findAllWithKV
    [ Se.And
        [ Se.Is BeamDF.endTime $ Se.GreaterThanOrEq from,
          Se.Is BeamDF.endTime $ Se.LessThanOrEq to,
          Se.Is BeamDF.feeType $ Se.Eq feeType,
          Se.Is BeamDF.merchantId $ Se.Eq merchantId.getId
        ]
    ]
```

Reply:
After that I will review

Reply:
changed for in case of findAlloptions and in case multiple conditions